jax/numpy/lax_numpy.py

```diff
@@ -979,6 +979,32 @@ def nan_reduction(a, axis=None, out=None, keepdims=False, **kwargs):
 nansum = _make_nan_reduction(onp.nansum, sum, 0, nan_if_all_nan=False)
 nanprod = _make_nan_reduction(onp.nanprod, prod, 1, nan_if_all_nan=False)
 
+@_wraps(onp.nanmean)
+def nanmean(a, axis=None, dtype=None, out=None, keepdims=False):
+  if out is not None:
+    raise ValueError("nanmean does not support the `out` argument.")
+  # Check and Count all NaN's per axis
+  nan_mask = isnan(a)
+  nan_cnt = sum(nan_mask, axis=axis, dtype=bool_, keepdims=keepdims)
+  if np.count_nonzero(nan_cnt) == 0:
```

Comment:
Second: using a Python branch on a value like `np.count_nonzero` won't work inside a `jit` decorator, because the value of `a` will be abstracted to a ShapedArray. Branches on abstract values currently have to be done using `lax.select`/`np.where` or `lax.cond`. But you don't need to branch here at all — just follow the algorithm I outlined above.

Reply:
Whoops on the NP front I implemented this function originally in a notebook for my own purposes, and forgot to remove them all. Interesting on the other front, I tested this a good amount but not wrapped in JIT, so thanks for informing me of those side-effects!